libraries/cbor/src/macros.rs

```diff
@@ -41,20 +41,31 @@ macro_rules! cbor_map_options {
     };
 
     ( $( $key:expr => $value:expr ),* ) => {
+        {
+            let mut _map = ::alloc::collections::BTreeMap::<_, $crate::values::Value>::new();
+            extend_cbor_map_options! ( &mut _map, $( $key => $value, )* );
+            $crate::values::Value::Map(_map)
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! extend_cbor_map_options {
+    // Add trailing comma if missing.
+    ( &mut $initial:expr, $( $key:expr => $value:expr ),+ ) => {
+        extend_cbor_map_options! ( &mut $initial, $($key => $value,)+ )
+    };
+
+    ( &mut $initial:expr, $( $key:expr => $value:expr, )* ) => {
```

Comment:
This should be named `$map` instead of `$initial`. This only contains the initial map at the beginning, then it's mutated.

Reply:
Done.